### DISCUSSION TITLE
sshj Executes the command

### DISCUSSION THREAD

**Discussion by wzd-hash** [hierynomus/sshj#869] [Q&A]
Hi, I have a question. I use SSHJ execute the command "nohup Java jar - server - Xms1g - Xmx1g - XX: XX: + UseG1GC - + UnlockExperimentalVMOptions - XX: G1NewSizePercent = 20 -XX:G1MaxNewSizePercent=30 -XX:+DisableExplicitGC /data/deploy/jar/nut-helloworld.jar --spring.profiles.active=dev >> /data/deploy/jar/nut-helloworld.log 2> &1&", returns a status code of 0, but it does not take effect. The same command that I wrote in the script worked when the script was executed

The code is as follows：

public static SSHClient connect(String host, Integer port, String user, String password) throws IOException {
        SSHClient ssh = new SSHClient();
        ssh.addHostKeyVerifier(new PromiscuousVerifier());
        ssh.connect(host,port);
        ssh.authPassword(user, password);
        return ssh;
    }

    public static Result execCommand(SSHClient ssh, String command,boolean isSudo){
        Result r = new Result();
        Session session = null;
        try {
            session = ssh.startSession();
            command = getCommand(command, isSudo);
            final Session.Command cmd = session.exec(command);
            cmd.join(TIMEOUT, TimeUnit.SECONDS);
            String runLog = IOUtils.readFully(cmd.getInputStream()).toString();
            String errorLog = IOUtils.readFully(cmd.getErrorStream()).toString();
            Integer exitStatus = cmd.getExitStatus();
            r = handleResult(exitStatus, runLog, errorLog, command);
        } catch (ConnectionException e) {
            r.setStatus(-1);
            r.handleErrorLog(e.getMessage());
            r.setIsSuccess(false);
        } catch (IOException e) {
            r.setStatus(-1);
            r.handleErrorLog(e.getMessage());
            r.setIsSuccess(false);
        } finally {
            closeSession(session);
        }

        return r;
    }

**Comment by hierynomus** [comment]
If you background the process (the last `&`), the return code will be `0` as the process will not be executed attached and the ssh command will immediately terminate.
